lib/resty/websocket/client.lua

```diff
@@ -98,9 +98,17 @@ function _M.connect(self, uri, opts)
         path = "/"
     end
 
-    local ssl_verify, proto_header, origin_header, sock_opts = false
+    local ssl_verify, proto_header, origin_header, custom_header, sock_opts = false
 
     if opts then
+        local headers = opts.headers
+        if headers then
+            custom_header = ""
+            for field,value in pairs(headers) do
+                custom_header = custom_header .. "\r\n" .. field .. ": " .. value
```

Comment:
Indeed, however is not like this is a critical code, neither many of custom headers usually will be used and I see no guaranteed performant way to avoid concacenating with "..". 

We could use string.format() in each iteration here, but is not clear if it would perform better. We could also create an auxiliary table and use table.concat(), but again is not clear if creating another table to do that task would perform better.

Anyway I have no intention to spend time micro optimazing here as this is not a critical code.